src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -1420,16 +1458,20 @@ nns_edge_set_info (nns_edge_h edge_h, const char *key, const char *value)
     SAFE_FREE (eh->ip);
     eh->ip = nns_edge_strdup (value);
   } else if (0 == strcasecmp (key, "PORT")) {
-    eh->port = g_ascii_strtoll (value, NULL, 10);
+    eh->port = (int) strtoll (value, NULL, 10);
   } else if (0 == strcasecmp (key, "TOPIC")) {
     SAFE_FREE (eh->topic);
     eh->topic = nns_edge_strdup (value);
+  } else if (0 == strcasecmp (key, "ID")) {
+    nns_edge_loge ("Cannot update ID.");
+    ret = NNS_EDGE_ERROR_INVALID_PARAMETER;
```

Comment:
set the client id if IP is valid?

Reply:
Update src for 'ID' and 'Client_ID':
client-ID is internal, and ID for handle should be set when creating edge handle.